content/blog/zephyr-tracing-and-profiling/index.md

```diff
@@ -0,0 +1,279 @@
+﻿---
+title: "Introduction to Zephyr OS Tracing and Profiling"
+date: 2024-10-25T06:54:17+08:00
+showAuthor: false
+featureAsset: "featured-gauge.webp"
+authors:
+  - "raffael-rostagno"
+tags: ["ESP32", "Tracing",  "Profiling","Zephyr"]
+---
+
+## Overview
+
+Embedded systems can be quite complex, depending on the architecture, application size and nature. Even for an experienced embedded developer, understanding the interplay of threads, interrupts and many processes that run in a large application can be challenging in a situation where debugging, evaluating hardware adequacy or optimizing application code is necessary. We humans are (mostly) visual by nature, and having the means to *visualize* what is happening in a given system can really open up possibilities. Without the right tools we are often in the dark, quite literally.

```

Comment:
`interrupts and many processes that run in a large application can be challenging in a situation where debugging, evaluating hardware adequacy or optimizing application code is necessary.`
This part sounds too large. `interrupts and many processes that run in a large application can be challenging` could be the end of the sentence.